tests/run-make/output-type-permutations/rmake.rs

```diff
@@ -0,0 +1,177 @@
+    fs_wrapper::remove_file(&bin_foo);
+    rustc().input("foo.rs").emit(&format!("link={bin_foo}")).run();
+    fs_wrapper::remove_file(&bin_foo);
+    rustc().input("foo.rs").arg(&format!("--emit=link={bin_foo}")).run();
+    fs_wrapper::remove_file(&bin_foo);
+    std::fs::remove_file("foo.pdb").unwrap_or_default();
+    assert_eq!(fs_wrapper::read_dir(cwd()).count(), 1);
+
+    rustc().input("foo.rs").crate_type("rlib").emit("link").output("foo").run();
+    fs_wrapper::remove_file("foo");
+    rustc().input("foo.rs").crate_type("rlib").emit("link=foo").run();
+    fs_wrapper::remove_file("foo");
+    rustc().input("foo.rs").crate_type("rlib").arg("--emit=link=foo").run();
+    fs_wrapper::remove_file("foo");
+    assert_eq!(fs_wrapper::read_dir(cwd()).count(), 1);
+
+    rustc().input("foo.rs").crate_type("dylib").emit("link").output("foo").run();
+    fs_wrapper::remove_file("foo");
+    rustc().input("foo.rs").crate_type("dylib").emit("link=foo").run();
+    fs_wrapper::remove_file("foo");
+    rustc().input("foo.rs").crate_type("dylib").arg("--emit=link=foo").run();
+    fs_wrapper::remove_file("foo");
+    remove_artifacts();
+    assert_eq!(fs_wrapper::read_dir(cwd()).count(), 1);
+
+    rustc().input("foo.rs").crate_type("staticlib").emit("link").output("foo").run();
+    fs_wrapper::remove_file("foo");
+    rustc().input("foo.rs").crate_type("staticlib").emit("link=foo").run();
+    fs_wrapper::remove_file("foo");
+    rustc().input("foo.rs").crate_type("staticlib").arg("--emit=link=foo").run();
+    fs_wrapper::remove_file("foo");
+    assert_eq!(fs_wrapper::read_dir(cwd()).count(), 1);
+
+    let bin_foo = bin_name("foo");
+    rustc().input("foo.rs").crate_type("bin").emit("link").output(&bin_foo).run();
+    fs_wrapper::remove_file(&bin_foo);
+    rustc().input("foo.rs").crate_type("bin").emit(&format!("link={bin_foo}")).run();
+    fs_wrapper::remove_file(&bin_foo);
+    rustc().input("foo.rs").crate_type("bin").arg(&format!("--emit=link={bin_foo}")).run();
+    fs_wrapper::remove_file(&bin_foo);
+    std::fs::remove_file("foo.pdb").unwrap_or_default();
+    assert_eq!(fs_wrapper::read_dir(cwd()).count(), 1);
+
+    rustc().input("foo.rs").emit("llvm-ir=ir").emit("link").crate_type("rlib").run();
+    fs_wrapper::remove_file("ir");
+    fs_wrapper::remove_file(rust_lib_name("bar"));
+    assert_eq!(fs_wrapper::read_dir(cwd()).count(), 1);
+
+    rustc()
+        .input("foo.rs")
+        .emit("asm=asm")
+        .emit("llvm-ir=ir")
+        .emit("llvm-bc=bc")
+        .emit("obj=obj")
+        .emit("link=link")
+        .crate_type("staticlib")
+        .run();
+    fs_wrapper::remove_file("asm");
+    fs_wrapper::remove_file("ir");
+    fs_wrapper::remove_file("bc");
+    fs_wrapper::remove_file("obj");
+    fs_wrapper::remove_file("link");
+    assert_eq!(fs_wrapper::read_dir(cwd()).count(), 1);
+
+    rustc()
+        .input("foo.rs")
+        .arg("--emit=asm=asm")
+        .emit("llvm-ir=ir")
```

Comment:
I would use manual `arg` here, to make it clear that the difference is between `emit x` and `emit=x`.